### PR TITLE
EVG-12929 create spruce config resolver with uservoice

### PR DESCRIPTION
**Pull request by khelif96**
This creates a `spruceConfig` resolver which will contain all general config options. I am migrating the banner resolver into this as well as adding a `uservoice` resolver which will contain a link to `uservoice`

Support for https://github.com/evergreen-ci/spruce/pull/405

### CODE REVIEW

**Comment by SupaJoon** [comment]
why not map APIAdminSettings to the SpruceConfig gql type in `gqlgen.yml`? now might be a good chance since are already changing the the gql endpoint for the banner theme

File: testdata/local/admin.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{ "_id" : "ui", "default_project" : "evergreen", "url" : "localhost:9090", "http_listen_addr" : ":9090", "secret" : "this is a secret", "cors_origins": ["http://localhost:3000"] }
+{ "_id" : "ui", "default_project" : "evergreen", "url" : "localhost:9090", "http_listen_addr" : ":9090", "secret" : "this is a secret", "cors_origins": ["http://localhost:3000"], "user_voice": "https://feedback.mongodb.com/forums/930019-evergreen"}
```
Comment (john-m-liu) [inline testdata/local/admin.json:1]: Can you change this to a fake url?

File: config_ui.go [hunk 2]
```diff
@@ -23,6 +23,7 @@ type UIConfig struct {
 	CORSOrigins             []string `bson:"cors_origins" json:"cors_origins" yaml:"cors_origins"`         // allowed request origins for some UI Routes
 	LoginDomain             string   `bson:"login_domain" json:"login_domain" yaml:"login_domain"`         // domain for the login cookie (defaults to domain of app)
 	ExpireLoginCookieDomain string   `bson:"expire_domain" json:"expire_domain" yaml:"expire_domain"`      // expire login token for this domain when the user logs in
+	UserVoice               string   `bson:"user_voice" json:"user_voice" yaml:"user_voice"`
```
Comment (john-m-liu) [inline config_ui.go:26]: super nit but I think UserVoice is just 1 word, so I would remove the underscore in these tags as well as in the rest model

**Review by john-m-liu — CHANGES_REQUESTED** [review]

File: graphql/resolvers.go [hunk 3]
```diff
@@ -1402,6 +1403,26 @@ func (r *queryResolver) SiteBanner(ctx context.Context) (*restModel.APIBanner, e
...
+	err = spruceConfig.BuildFromService(config)
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error building api admin settings from service: %s", err.Error()))
+	}
+	uiConfig := restModel.APIUIConfig{}
+	err = uiConfig.BuildFromService(config.Ui)
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error building api UI Config from service: %s", err.Error()))
+	}
+	spruceConfig.Ui = &uiConfig
```
Comment (SupaJoon) [inline graphql/resolvers.go:1422]: it is also possible to do this mapping in APIAdminSettings.BuildFromService. I'm not sure if doing that would have an additional benefit so I think it's fine here

**Review by SupaJoon — APPROVED** [review]

File: graphql/resolvers.go (hunk 3 above)
Comment (john-m-liu) [inline graphql/resolvers.go:1422]: I don't believe you need to do this - calling BuildFromService on the config variable should populate this already

Comment (khelif96) [reply graphql/resolvers.go]: It wasnt populating but i just updated buildFromService to build uiconfig

**Review by john-m-liu — APPROVED** [review]

**Pull request merged**